### PR TITLE
#1270 hints for new users

### PR DESCRIPTION
**Pull request by Sibgha360**
**What's in the PR**
* Hints for new users

**How to test manually**
* Pull these webanno PRs: https://github.com/webanno/webanno/pull/1584, https://github.com/webanno/webanno/pull/1590
* Open the website for the first time, there's now a *Start Tutorial* button.
* Click this button and walk through the tutorial.

* Test that other project functionality still works after tutorial ended/was skipped.

**Documentation**
* [ ] PR updates documentation

### CODE REVIEW

**Comment by UWinch** [comment]
Jenkins, can you test this please?

File: inception-ui-core/pom.xml [hunk 1]
```diff
@@ -164,6 +164,12 @@
       <groupId>de.tudarmstadt.ukp.inception.app</groupId>
       <artifactId>inception-support</artifactId>
     </dependency>
+	<dependency>
+	  <groupId>javax.servlet</groupId>
```
Comment (jcklie) [inline inception-ui-core/pom.xml]: This is an old version and wrong (?) artifact id (javax.servlet-api)

**Comment by UWinch** [comment]
Jenkins, can you test this please?

**Comment by UWinch** [comment]
> I tried to walk through the whole tutorial a couple of times but could not finish and could not do certain steps. I used Firefox.
> 
> What I like:
> 
>     * Style and design
> 
>     * hints have clear an short texts
> 
>     * I like the simplicity! I think it's a good thing and well made.
> 
> 
> What could be better:
> 
>     * Instead of "skip"-button: "Exit tutorial"/ "Exit hints"/"Exit tour"/"Exit" or the like. Reason: "skip" makes me think of skipping this step, so I need to think about wether to klick "skip" or "next" for this.
> 
>     * Step "Evaluation": The hint says: "This allows....recommenders...". Bit Users don't yet know what a recommender is. I suggest: "This allows....recommenders.... (A recommender is...)"
> 
>     * When I quit, I have to start with a new Tutorial-Project. This is not so important because main reason for starting new was I had to cancel due to Problems to be solved (see below).
> 
> 
> What should be better:
> 
>     * Have a "back"-button which gets you one step back without exiting. Reason: I wanted to change my filter from "Manager" to "Annotator" but could not withour starting anew. Also, maybe you want to look something up again quickly.
> 
>     * Have the the hints in the order of a standard workflow (I guess, "Setup" first?). Reason: for example the steps "annotate" (also: "Curation", "Agreement", "Monitoring", "Evaluate"),: I have nothing to annotate/curate (yet).
> 
>     * Have a "back"-button entegrated when we are on sub-pages. Reason: When I am in step "Annotate" (also: "Curation", "Agreement", "Monitoring", "Evaluate"), I cannot get back to the main page. Using the back-arrow of my browser, th esteps of this page start anew.
> 
>     * Step Setup: The Button "Next" and "Exit (a small cross)" are layerd on each other, interfering with each other AND the standard layer below. The should be sepereate and easy to see.
> 
>     * Step Setup: When I click in Setup, entering the subpage, I cannot do anythin anymore. All is greyed out. By going one step back by the browsers back-arrow, I quit the tutorial. When getting on SetUp again, still everythin is greyed out except for "Next" and "Exit" as usual.

Thanks, I will change some of the order of the tutorial. 
There seems to also still be a bug that I need to fix when parts of the page are not loaded yet (the problems you had in the setup part). 
And I think there was some confusion when I wanted you to click on a button and when I just wanted to tell you about what the button does in general (the buttons for `Annotation`, `Curation` etc.). Maybe, I could fix this by re-phrasing the text?

File: inception-ui-core/src/main/java/de/tudarmstadt/ukp/inception/ui/core/footer/resources/tutorial.js [hunk 2]
```diff
@@ -427,15 +486,15 @@ function goBackToDashboardRoutine()
 	enjoyhint_instance.set(
 	[
 		{
-			'click [href=\'./project.html\']:last' : 'Now, lets go to the Dashboard.',
+			'click [href=\'./project.html\']:last' : 'Now, lets go back to the Dashboard.',
```
Comment (UWinch) [inline inception-ui-core/src/main/java/de/tudarmstadt/ukp/inception/ui/core/footer/resources/tutorial.js]: goBackToDashboard function is not used anymore

**Comment by UWinch** [comment]
@An-Feli 
Please check again if this now works better for you. 
* I hopefully fixed the problem with being stuck during the tutorial at a weird overlay of buttons.
* I did not implement a back button or change the `Skip` button because the used library does not easily allow it. Let me know if this is a fatal flaw.
* I disabled some buttons during the tutorial to make the path through it more clear.
* I changed the order of explanations and setup.

**Comment by UWinch** [comment]
jenkins, can you test thi please?

**Comment by UWinch** [comment]
Jenkins, can you test this please?

**Comment by jcklie** [comment]
@An-Feli Can you have another look please?

**Comment by JackDapid** [comment]
Sorry if obvious, 
I can't find the tutorial, tried to create new users, login with them, new or private browser and can't see any tutorial - is this after 0.14.2? Is there a link to open the tutorial later?

**Comment by UWinch** [comment]
Hi @JackDapid , the tutorial is not merged into master or released yet. If you want to test it, you need to checkout the current webanno master branch, as well as this INCEpTION branch feature/1270-hints-for-new-users and build the project yourself. 
Then you should see a *Start Tutorial* link on the Projects Overview page.
I would be happy about your feedback before we release the feature, but you could also wait for the 0.15.0 release to try it out.

**Comment by JackDapid** [comment]
Thanks @UWinch trying now to give feedback & since first time building it myself might have done something wrong:
1) git clone webanno, 'mvn install' -> all good
2) git clone inception, 'git checkout feature/1270-hints-for-new-users', 'mvn install' -> lapps testing error

`[INFO] Running de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest
[ERROR] Tests run: 3, Failures: 0, Errors: 2, Skipped: 1, Time elapsed: 66.39 s <<< FAILURE! - in de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest
[ERROR] testNerConformity(LappsGridService{name='ABNER 1.0.0', description='ABNER', url='http://vassar.lappsgrid.org/invoker/anc:gate.abner_1.0.0'}) [0](de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest)  Time elapsed: 34.18 s  <<< ERROR!
de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommendationException: Cannot predict
	at de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest.predict(LappsGridRecommenderConformityTest.java:112)
	at de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest.testNerConformity(LappsGridRecommenderConformityTest.java:73)
Caused by: java.lang.RuntimeException: <html><head><title>Apache Tomcat/6.0.35 - Error report</title><style></style> </head><body><h1>HTTP Status 408 - </h1><HR size="1" noshade="noshade"><p><b>type</b> Status report</p><p><b>message</b> <u></u></p><p><b>description</b> <u>The client did not produce a request within the time that the server was prepared to wait ().</u></p><HR size="1" noshade="noshade"><h3>Apache Tomcat/6.0.35</h3></body></html>
	at de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest.predict(LappsGridRecommenderConformityTest.java:112)
	at de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest.testNerConformity(LappsGridRecommenderConformityTest.java:73)

[ERROR] testNerConformity(LappsGridService{name='edu.cmu.lti.oaqa.lapps.LingpipeNER - 1.1.1-SNAPSHOT', description='Lingpipe Named Entity Recognizer with model "English News: MUC-6"', url='http://vassar.lappsgrid.org/invoker/anc:lingpipe.ner_1.1.1-SNAPSHOT'}) [1](de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest)  Time elapsed: 31.746 s  <<< ERROR!
de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommendationException: Cannot predict
	at de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest.predict(LappsGridRecommenderConformityTest.java:112)
	at de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest.testNerConformity(LappsGridRecommenderConformityTest.java:73)
Caused by: java.lang.RuntimeException: <html><head><title>Apache Tomcat/6.0.35 - Error report</title><style></style> </head><body><h1>HTTP Status 408 - </h1><HR size="1" noshade="noshade"><p><b>type</b> Status report</p><p><b>message</b> <u></u></p><p><b>description</b> <u>The client did not produce a request within the time that the server was prepared to wait ().</u></p><HR size="1" noshade="noshade"><h3>Apache Tomcat/6.0.35</h3></body></html>
	at de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest.predict(LappsGridRecommenderConformityTest.java:112)
	at de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsGridRecommenderConformityTest.testNerConformity(LappsGridRecommenderConformityTest.java:73)

[INFO] Running de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsRecommenderIntegrationTest
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0 s - in de.tudarmstadt.ukp.inception.recommendation.imls.lapps.LappsRecommenderIntegrationTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   LappsGridRecommenderConformityTest.testNerConformity:73->predict:112 » Recommendation
[ERROR]   LappsGridRecommenderConformityTest.testNerConformity:73->predict:112 » Recommendation
[INFO] 
[ERROR] Tests run: 4, Failures: 0, Errors: 2, Skipped: 2
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for INCEpTION 0.15.0-SNAPSHOT:
[INFO] 
[INFO] INCEpTION .......................................... SUCCESS [  3.952 s]
[INFO] INCEpTION - Build resources ........................ SUCCESS [  1.302 s]
[INFO] INCEpTION - Plugin Parent POM ...................... SUCCESS [  0.566 s]
[INFO] INCEpTION - Log .................................... SUCCESS [ 10.954 s]
[INFO] INCEpTION - Support library ........................ SUCCESS [  2.050 s]
[INFO] INCEpTION - Scheduling ............................. SUCCESS [  3.759 s]
[INFO] INCEpTION - Recommendation - API ................... SUCCESS [  3.381 s]
[INFO] INCEpTION - Test Support library ................... SUCCESS [  9.247 s]
[INFO] INCEpTION - UI - Core .............................. SUCCESS [  2.153 s]
[INFO] INCEpTION - Documentation .......................... SUCCESS [  0.346 s]
[INFO] INCEpTION - Document Metadata Support .............. SUCCESS [  1.505 s]
[INFO] INCEpTION - Image Support .......................... SUCCESS [  1.448 s]
[INFO] INCEpTION - Knowledge Base - Core .................. SUCCESS [03:11 min]
[INFO] INCEpTION - Concept Linking ........................ SUCCESS [ 35.704 s]
[INFO] INCEpTION - Search - Core .......................... SUCCESS [  3.273 s]
[INFO] INCEpTION - Search - MTAS .......................... SUCCESS [ 48.111 s]
[INFO] INCEpTION - Knowledge Base - UI .................... SUCCESS [ 17.278 s]
[INFO] INCEpTION - Recommendation - Core .................. SUCCESS [ 18.015 s]
[INFO] INCEpTION - ML - String Matching ................... SUCCESS [ 33.916 s]
[INFO] INCEpTION - ML - External .......................... SUCCESS [ 23.032 s]
[INFO] INCEpTION - ML - LAPPS Grid ........................ FAILURE [01:16 min]
[INFO] INCEpTION - ML - OpenNLP (v 1.9.1) ................. SKIPPED
[INFO] INCEpTION - ML - WebLicht .......................... SKIPPED
[INFO] INCEpTION - Recommendation - Active Learning ....... SKIPPED
[INFO] INCEpTION - Search - UI ............................ SKIPPED
[INFO] INCEpTION - External Search - Core ................. SKIPPED
[INFO] INCEpTION - External Search - UI ................... SKIPPED
[INFO] INCEpTION - External Search - Elastic Search ....... SKIPPED
[INFO] INCEpTION - External Search - PubAnnotation ........ SKIPPED
[INFO] INCEpTION - Editor - HTML .......................... SKIPPED
[INFO] INCEpTION - Editor - PDF ........................... SKIPPED
[INFO] INCEpTION - Web-Application ........................ SKIPPED
[INFO] INCEpTION - ML - Deeplearning4J (v1.0.0-beta6) ..... SKIPPED
[INFO] INCEpTION - Example - ML - Data Majority Recommender SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  08:08 min
[INFO] Finished at: 2020-01-28T15:00:10+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project inception-imls-lapps: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/ddobos/Desktop/INCEPTION/inception/inception-imls-lapps/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :inception-imls-lapps`

**Comment by jcklie** [comment]
I would skip running the tests by issuing

` mvn -U clean install -DskipTests  -D"checkstyle.skip"=true -D"mdep.analyze.skip"=true`

**Comment by JackDapid** [comment]
Really nice feature, just minor comments from me:
- The always present skip button made me stop the tutorial by mistake several times (is the exit button on the top right not enough to end the tutorial) ... but having a real skip functionality to skip to the next tutorial point (if for example I don't want to create the 5th project) would be really nice - otherwise agree to An-Feli renaming (and maybe even re-coloring) would be good
- if the text wraps into two lines it overlaps with the buttons (no idea if avoidable in the current implementation) (see below in screenshot)
- This will be most useful for tutorials for annotators - so instead or additional of the trainings I have to currently do with them - so how can I help?

Screenshot 2020-01-28 at 17 27 22

**Comment by JackDapid** [comment]
Just another quick finding/issue:
- if a user without project-creator or admin role runs the tutorial - it fails at the project creation step (as it can't highlight the project creation but highlights Annotator - see screenshot below) 

Screenshot 2020-01-28 at 18 07 57

**Comment by UWinch** [comment]
@JackDapid @An-Feli thanks a lot for both of your in-depth feedback, this is really useful. I will work it into the tutorial for the coming release.

**Comment by UWinch** [comment]
Jenkins can you test this please?

**Comment by UWinch** [comment]
Jenkins can you test this please?

**Pull request merged**